src/components/carousel/carousel.js

```diff
@@ -92,11 +138,46 @@ class Carousel extends React.Component {
    * @method componentWillMount
    */
   componentWillMount() {
-    const selectedIndex = Number(this.props.initialSlideIndex);
+    const selectedIndex = Number(this.props.slideIndex) || Number(this.props.initialSlideIndex);
     this.setState({ selectedSlideIndex: selectedIndex });
   }
 
   /**
+   * A lifecycle method that is called before re-render.
+   *
+   * @method componentWillReceiveProps
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.slideIndex === undefined) { return; }
```

Comment:
it's generally better to do `typeof foo === 'undefined'`.
If `foo` is not declared, the `foo === undefined` would throw an error.